subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c

```diff
@@ -234,6 +237,72 @@ static int handle_device_config_update(const struct nct_evt *const evt,
 	return err;
 }
 
+/* Placeholder until cloud logging added in another PR */
+void nrf_cloud_log_control(int log_level)
+{
+	LOG_DBG("Setting nRF Cloud log level = %d", log_level);
+}
+
+static int handle_device_control_update(const struct nct_evt *const evt,
+					bool *const control_found)
+{
+	int err;
+	struct nct_cc_data msg = {
+		.opcode = NCT_CC_OPCODE_UPDATE_REQ,
+		.message_id = NCT_MSG_ID_STATE_REPORT,
+	};
+	struct nrf_cloud_evt cloud_evt = {
+		.type = NRF_CLOUD_EVT_RX_DATA_SHADOW
+	};
+	enum nrf_cloud_ctrl_status status = NRF_CLOUD_CTRL_NOT_PRESENT;
+	struct nrf_cloud_ctrl_data ctrl_data;
+
+	if ((evt == NULL) || (control_found == NULL)) {
+		return -EINVAL;
+	}
+
+	if (evt->param.cc == NULL) {
+		return -ENOENT;
+	}
+
+	err = nrf_cloud_decode_control(&evt->param.cc->data, &status, &ctrl_data);
+	if (err) {
+		return (err == -ESRCH) ? 0 : err;
+	}
+
+	*control_found = (status != NRF_CLOUD_CTRL_NOT_PRESENT);
+	if (*control_found) {
+		nrf_cloud_alert_control(ctrl_data.alerts_enabled);
+		nrf_cloud_log_control(ctrl_data.log_level);
+	}
+
+	if (status != NRF_CLOUD_CTRL_REPLY) {
+		return 0;
```

Comment:
This will bypass the call to `nfsm_set_current_state_and_notify(nfsm_get_current_state(), &cloud_evt);`, though IDK if that call is necessary -- is it?

Reply:
A comment indicating the purpose of that call would probably be a good idea either way -- why does the fsm need to re-enter its current state in response to a control update?

And, if intentional, why does this only need to happen when nrf_cloud needs a response from us?